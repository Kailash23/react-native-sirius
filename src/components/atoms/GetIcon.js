import React from 'react';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import Feather from 'react-native-vector-icons/Feather';
import { withTheme } from 'react-native-paper';

function GetIcon(props) {
  const { colors } = props.theme;
  if (props.isMaterial) {
    return (
      <MaterialCommunityIcons
        name={props.name}
        size={props.size}
        color={props.color}
        style={props.style || styles}
      />
    );
  } else if (props.isFeather) {
    return (
      <Feather
        name={props.name}
        size={props.size}
        color={props.color}
        style={props.style || styles}
      />
    );
  } else {
    return (
      <Ionicons
        name={props.name}
        size={props.size}
        color={props.color}
        style={props.style || styles}
      />
    );
  }
}
const styles = {};

export default withTheme(GetIcon);
