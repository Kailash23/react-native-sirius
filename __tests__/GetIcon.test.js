import GetIcon from '@atoms/GetIcon';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
const moment = require('moment');

configure({ adapter: new Adapter() });

describe('Testing <GetIcon/>', () => {
  test('GetIcon should return white Meterial close icon of size 45 ', () => {
    const wrapper = shallow(
      <GetIcon isMaterial name={'close'} size={45} color={'white'} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test('GetIcon should return black isFeather arrow down icon of size 20 ', () => {
    const wrapper = shallow(
      <GetIcon isFeather name={'arrow-down-left'} size={20} color={'black'} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
