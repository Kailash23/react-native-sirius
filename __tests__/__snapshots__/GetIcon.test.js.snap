// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GetIcon/> GetIcon should return black isFeather arrow down icon of size 20  1`] = `
<ThemedComponent
  _reactThemeProviderForwardedRef={null}
  color="black"
  isFeather={true}
  name="arrow-down-left"
  size={20}
/>
`;

exports[`Testing <GetIcon/> GetIcon should return white Meterial close icon of size 45  1`] = `
<ThemedComponent
  _reactThemeProviderForwardedRef={null}
  color="white"
  isMaterial={true}
  name="close"
  size={45}
/>
`;
